dm/syncer/metrics/metrics.go

```diff
@@ -30,302 +31,350 @@ const (
 	BinlogEventCostStageGenQuery      = "gen-query"
 )
 
-// below variables are exported to syncer package.
-var (
-	BinlogReadDurationHistogram = metricsproxy.NewHistogramVec(
+// Metrics groups syncer's metric variables.
+type Metrics struct {
+	BinlogReadDurationHistogram     prometheus.Observer
+	BinlogEventSizeHistogram        prometheus.Observer
+	ConflictDetectDurationHistogram prometheus.Observer
+	IdealQPS                        prometheus.Gauge
+	BinlogMasterPosGauge            prometheus.Gauge
+	BinlogSyncerPosGauge            prometheus.Gauge
+	BinlogMasterFileGauge           prometheus.Gauge
+	BinlogSyncerFileGauge           prometheus.Gauge
+	BinlogEventRowHistogram         prometheus.Observer
+	TxnHistogram                    prometheus.Observer
+	QueryHistogram                  prometheus.Observer
+	SyncerExitWithErrorCounter      prometheus.Counter
+	ReplicationLagGauge             prometheus.Gauge
+	ReplicationLagHistogram         prometheus.Observer
+	RemainingTimeGauge              prometheus.Gauge
+	ShardLockResolving              prometheus.Gauge
+	FinishedTransactionTotal        prometheus.Counter
+	FlushCheckPointsTimeInterval    prometheus.Observer
+}
+
+// Proxies provides the ability to clean Metrics values when syncer is closed.
+type Proxies struct {
+	Metrics                         *Metrics
+	BinlogReadDurationHistogram     *metricsproxy.HistogramVecProxy
```

Comment:
it seems that only `Metrics` is accessed by callers, maybe make other members like  `BinlogReadDurationHistogram` to private members?

Reply:
some variables like UnsyncedTableGauge are accessed in syncer package, I'll make rest of them private.